src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java

```diff
@@ -474,27 +493,101 @@ private void mapVariableToTaskResourceProperty(TaskResource taskResource, String
                     }
                     break;
                 case MINOR_PRIORITY:
-                    if (value instanceof String) {
-                        taskResource.setMinorPriority(Integer.parseInt((String) value));
-                    } else {
-                        taskResource.setMinorPriority((Integer) value);
-                    }
-
+                    setMinorPriority(value, taskResource);
                     break;
                 case MAJOR_PRIORITY:
+                    setMajorPriority(value, taskResource);
+                    break;
+                case PRIORITY_DATE:
                     if (value instanceof String) {
-                        taskResource.setMajorPriority(Integer.parseInt((String) value));
+                        taskResource.setPriorityDate(OffsetDateTime.parse((String) value));
                     } else {
-                        taskResource.setMajorPriority((Integer) value);
+                        taskResource.setPriorityDate((OffsetDateTime) value);
                     }
                     break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    void setMajorPriority(Object value, TaskResource taskResource) {
+        if (value instanceof String) {
+            taskResource.setMajorPriority(Integer.parseInt((String) value));
+        } else {
+            taskResource.setMajorPriority((Integer) value);
+        }
+    }
+
+    void setMinorPriority(Object value, TaskResource taskResource) {
+        if (value instanceof String) {
+            taskResource.setMinorPriority(Integer.parseInt((String) value));
+        } else {
+            taskResource.setMinorPriority((Integer) value);
+        }
+    }
+
+    protected void reconfigureTaskAttribute(TaskResource taskResource,
```

Comment:
reconfigureTaskAttribute method seems to have code duplicated from mapVariableToTaskResourceProperty. Is it possible to combine these two methods with check on reconfigure?

Reply:
they both have different set of properties, don't want to mess with existing logic